scripts/sanityCheck.js

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env babel-node
+
+import { Log, txUtils, env, eth, contracts, cli } from 'decentraland-commons'
+import { db } from '../src/database'
+import { Parcel, ParcelService } from '../src/Parcel'
+import { Publication } from '../src/Publication'
+import { BlockchainEvent } from '../src/BlockchainEvent'
+import { asyncBatch } from '../src/lib/asyncBatch'
+import { loadEnv } from './utils'
+import { decodeAssetId } from './monitor/utils'
+
+const log = new Log('update')
+
+let BATCH_SIZE
+
+export async function sanityCheck() {
+  log.info(`Using ${BATCH_SIZE} as batch size, configurable via BATCH_SIZE`)
+
+  const shouldFix = await cli.confirm('Fix errors?')
+
+  log.info('Connecting database')
+  await db.connect()
+
+  log.info('Connecting to Ethereum node')
+  await eth.connect({
+    contracts: [contracts.LANDRegistry, contracts.Marketplace],
+    providerUrl: env.get('RPC_URL')
+  })
+
+  log.info('[Check 1/2]: Checking Marketplace sanity against all events')
+  try {
+    await marketplaceCheck(shouldFix)
+  } catch (err) {
+    console.log(err, err.stack)
+  }
+
+  log.info('[Check 2/2]: Checking parcel ownership from -150,-150 to 150,150')
+  await processParcels(shouldFix)
+}
+
+async function marketplaceCheck(fix) {
+  const marketplace = eth.getContract('Marketplace')
+  const events = await getAllEvents(marketplace)
+  const filterParcels = getParcelIds(events)
+
+  const tasks = []
+  for (let d of filterParcels) {
+    tasks.push(checkParcel(d.parcelId, d.lastEvent, marketplace, fix))
+  }
+  console.log('Analyzing', tasks.length, 'cases of parcels')
+  await Promise.all(tasks)
+}
+
+function getAllEvents(marketplace) {
+  return new Promise((resolve, reject) => {
+    marketplace.instance
+      .allEvents({
+        fromBlock: 0,
+        toBlock: 'latest'
+      })
+      .get(function(error, events) {
+        if (error) {
+          return reject(error)
+        } else {
+          return resolve(events)
+        }
+      })
+  })
+}
+
+function getParcelIds(events) {
+  const results = {}
+  for (let event of events) {
+    const str = event.args.assetId
+    if (!str) continue
+    if (event.removed && results[str]) continue
+    results[str] = { parcelId: event.args.assetId, lastEvent: event }
+  }
+  return Object.values(results)
+}
+
+const NULL =
+  '0x0000000000000000000000000000000000000000000000000000000000000000'
+
+async function checkParcel(parcelId, lastEvent, marketplace, fix) {
+  const res = await decodeAssetId(parcelId)
+  const [x, y] = res.split(',')
+  const publication = await Publication.findInCoordinate(x, y)
+  const auction = await marketplace.instance.auctionByAssetId(parcelId)
```

Comment:
This `.instance` is not necessary! the contract infers the methods from the ABI :D